cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/DefaultEmitter.java

```diff
@@ -56,6 +58,11 @@ public void emitError(InvalidEntry<T> value) {
     return entryList.iterator();
   }
 
+  public Iterator<InvalidEntry<T>> getErrorIterator() {
+    // we have to return iterator of a copy of list, as we need reset the errorList between stages.
+    return new ArrayList<>(errorList).iterator();
```

Comment:
does this mean we're not immediately consuming them?  the normal iterator() does not return a copy.